cflib/localization/ippe_cf.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+#
+# ,---------,       ____  _ __
+# |  ,-^-,  |      / __ )(_) /_______________ _____  ___
+# | (  O  ) |     / __  / / __/ ___/ ___/ __ `/_  / / _ \
+# | / ,--'  |    / /_/ / / /_/ /__/ /  / /_/ / / /_/  __/
+#    +------`   /_____/_/\__/\___/_/   \__,_/ /___/\___/
+#
+# Copyright (C) 2022 Bitcraze AB
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, in version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+from collections import namedtuple
+
+import numpy as np
+import numpy.typing as npt
+
+from ._ippe import mat_run
+
+
+class IppeCf:
+    """
+    A wrapper class to simplify usage of IPPE in CF code.
+    Converts between CF style of coordinate systems/data structures and
+    IPPE (Open CV) style.
+    """
+
+    # Rotation matrix to transform from IPPE to CF
+    _R_ippe_to_cf = np.array([
+        [0.0, 0.0, 1.0],
+        [-1.0, 0.0, 0.0],
+        [0.0, -1.0, 0.0],
+    ])
+
+    # Rotation matrix to transform from CF to IPPE
+    _R_cf_to_ippe = np.transpose(_R_ippe_to_cf)
+
+    Solution = namedtuple('Solution', ['R', 't', 'reproj_err'])
+
+    @staticmethod
```

Comment:
Nice with types! Maybe we should use the renderable parameter description style in docstring though?

```
:param U_cf: Nx3 matrix holding the model points in world coordinates.
:param Q_cf: Nx2 matrix holding the points in the image. These are in normalised
            pixel coordinates. That is, the effects of the camera's intrinsic matrix
            and lens distortion are corrected, so that the Q projects with a perfect
            pinhole model.

:return: A list that contains 2 sets of pose solution from IPPE including rotation matrix
            translation matrix, and reprojection error. The first solution in the list has
            the smallest reprojection error.
```

Reply:
Agree! It was reused from the original ippe, my bad